app/services/migration/migrator.rb

```diff
@@ -49,12 +49,16 @@ def migrate(items, model)
       data_migration = DataMigration.find_by(model:)
       data_migration.update!(started_at: Time.zone.now, total_count: items.count)
 
-      items.each do |item|
+      failures_to_record = items.each_with_object([]) do |item, failures|
         data_migration.increment!(:processed_count)
         result = yield(item)
-        data_migration.increment!(:failure_count) unless result
+        unless result
+          data_migration.increment!(:failure_count)
+          failures << item
```

Comment:
Would this be simpler/easier to follow if we recorded on an item by item basis instead of collecting them? so:

```
data_migration.increment!(:failure_count)
failure_recorder.record_failure(data_migration:, item:)
```

Reply:
also to your question earlier Leandro, maybe we can also record the error message as well and pass it in with the item 🤔 otherwise will be quite hard to debug for now? let me know your thoughts